src/utilities/api/morg_http_client.py

```diff
@@ -148,7 +149,223 @@ def get_is_player_idle(self) -> Union[bool, None]:
                 print(e)
                 return None
 
-            if data["animation"] != -1 or data["animation pose"] not in [808, 813]:
+            if data["animation"] != animation_ids.IDLE or data["animation pose"] not in [808, 813]:
+                return False
+        return True
+    
+    def get_is_player_woodcutting(self) -> Union[bool, None]:
+        """
+        Checks if the player is doing a woodcutting animation.
+        Returns:
+                True if the player is woodcutting, False otherwise, or None if an error occurred.
+        """
+        # run a loop for 1 second
+        start_time = time.time()
+        while time.time() - start_time < 1:
```

Comment:
Could you, please, explain to me why this loop is necessary?

Reply:
I reused the code from the idle check, I just didn't want to change anything if it was necessary for some reason.

Reply:
@Mironor In these API plugins, there's no trustworthy property that tells us if our player is idle or not. Instead, we can check the current animation during a given tick. The problem is that as animations are happening, there is often an idle tick in between them (e.g., if you are in combat, between attacks you are considered idle). Therefore, we need to poll a few ticks and do a logical OR to determine if we are truly idle.